pkg/converter/cloudevents/cloudevents.go

```diff
@@ -60,10 +64,14 @@ func (ce *CloudEvent) Convert(data []byte) ([]byte, error) {
 	}
 
 	var body interface{}
+	contentType := "text/plain"
 	body = string(data)
 
-	// try to decode function's response into JSON
-	if json.Valid(data) {
+	// check if data is in JSON format
+	if json.Valid(data) &&
+		// "foo" is also JSON
+		bytes.Contains(data, []byte("{")) {
```

Comment:
If "foo" is _valid_ JSON, why would we restrict the cases to things which contain "{"? Even if it can be considered as either JSON or plain text, setting one value sounds better that not setting anything to me.
Using your logic, `"foo{bar}"` would trigger the switch, but it's still a string.

Reply:
If we _really_ want this, a better way would be

```go
bytes.TrimLeft(data)[0] == '{' || bytes.TrimLeft(data)[0] == '['
```

Reply:
Yeah, that's ugly. I just wanted to handle the cases when the function returns "foo" and runtime sets the` application/json` type. I can remove it but I'm sure that pretty soon we'll be asking ourselves why the functions with the simple string/int/etc. in responses considered as JSON.

Reply:
No I think that's a good idea :+1: It just needs more robust logic like the one I shared.

Reply:
Btw, why would a string be _quoted_ in the response of a function? Should it just respond this?

> foo

instead of

> "foo"

Reply:
I think it is either specific for the language runtime or for the API interaction. However, returned string values are quoted and I didn't find another solution but to add another string trimming.